MyApplication/app/src/main/java/com/justinwells/myapplication/MyMediaService.java

```diff
@@ -0,0 +1,94 @@
+package com.justinwells.myapplication;
+
+import android.app.IntentService;
+import android.app.Service;
+import android.content.Intent;
+import android.media.MediaPlayer;
+import android.os.Handler;
+import android.os.HandlerThread;
+import android.os.IBinder;
+import android.os.Looper;
+import android.os.Message;
+import android.support.annotation.Nullable;
+import android.widget.Switch;
+import android.widget.Toast;
+
+import java.io.IOException;
+
+/**
+ * Created by justinwells on 11/21/16.
+ */
+
+public class MyMediaService extends Service {
+    private MediaPlayer player = null;
+    private Handler handler;
+
+    @Nullable
+    @Override
+    public IBinder onBind(Intent intent) {
+        return null;
+    }
+
+    @Override
+    public void onCreate() {
+        super.onCreate();
+        player = MediaPlayer.create(MyMediaService.this, R.raw.rickastley_artists);
+
+        HandlerThread thread = new HandlerThread("MusicThread");
+        thread.start();
+
+        Looper looper = thread.getLooper();
+        handler = new Handler(looper) {
+            @Override
+            public void handleMessage(Message msg) {
+                super.handleMessage(msg);
+
+                String buttonPressed = (String) msg.obj;
+
+                switch (buttonPressed) {
+                    case MainActivity.PLAY_MUSIC:
+                        try {
+                            player.prepare();
```

Comment:
You should NOT call `prepare()` here, since you used the `MediaPlayer.create()` method, which delivers a prepared player. Calling this just crashes your app.

The documentation for `MediaPlayer.create()` says "Convenience method to create a MediaPlayer for a given resource id. On success, prepare() will already have been called and must not be called again."